dbms/src/Server/MetricsPrometheus.cpp

```diff
@@ -181,19 +195,12 @@ MetricsPrometheus::MetricsPrometheus(
             auto host = metrics_addr.substr(0, pos);
             auto port = metrics_addr.substr(pos + 1, metrics_addr.size());
 
-            auto service_addr = conf.getString("flash.service_addr");
-            std::string job_name = service_addr;
-            std::replace(job_name.begin(), job_name.end(), ':', '_');
-            std::replace(job_name.begin(), job_name.end(), '.', '_');
-            job_name = "tiflash_" + job_name;
-
-            char hostname[1024];
-            ::gethostname(hostname, sizeof(hostname));
-
-            gateway = std::make_shared<prometheus::Gateway>(host, port, job_name, prometheus::Gateway::GetInstanceLabel(hostname));
+            const String & job_name = "tiflash";
+            const auto & labels = prometheus::Gateway::GetInstanceLabel(conf.getString("flash.service_addr", getHostName()));
```

Comment:
What's the difference between the instance label here and the instance label in `InstanceLabelHolder`

Reply:
`InstanceLabelHolder` and `InstanceId` are both bad name...
InstanceLabelHolder has been renamed to `ComputeLabelHolder` and `ComputeLabelHolder::instance_id` has been renamed to `ComputeLabelHolder::process_id`.

Reply:
The `instance` label here is mean that the host_name of tiflash compute node.
The `process_id` label in `ComputeLabelHolder` is used to distinguish between processes that have been started multiple times.